types/dom.d.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Type declarations extending TypeScript's lib/lib.dom.d.ts.
+ * https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts
+ */
+
+interface Document {
+  readonly mozFullScreen: boolean;
+  readonly webkitIsFullScreen: boolean;
+
+  readonly fullscreenElement: Element;
+  readonly mozFullScreenElement: Element;
+  readonly msFullscreenElement: Element;
+  readonly webkitFullscreenElement: Element;
+
+  readonly mozFullScreenEnabled: boolean;
+  readonly msFullscreenEnabled: boolean;
+  readonly webkitFullscreenEnabled: boolean;
+
+  mozCancelFullScreen(): void;
+  msExitFullscreen(): void;
+  webkitExitFullscreen(): void;
+}
+
+interface Element {
+  mozRequestFullScreen(): Promise<void>;
+  mozRequestFullScreenWithKeys(): Promise<void>;
+  msRequestFullscreen(): Promise<void>;
+  webkitRequestFullscreen(allowKeyboardInput?: number): Promise<void>;
```

Comment:
I'm curious what makes these extend rather than override the existing types.  What would it look like if we wanted to override a type?

Reply:
These extend interfaces defined in the default library files loaded by `tsc` via TypeScript's [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html). To replace declarations, you would have to use the `--lib` option to `tsc` to override the default libs and omit those you want to replace. Docs on that [here](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

Reply:
Good to know.  Thanks @schmidtk.